### PR TITLE
feat: Add pickup in store support for GooglePay

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: CXSPA-5537

### CODE REVIEW

File: integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts [hunk 1]
```diff
@@ -187,61 +220,110 @@ export class OpfGooglePayService {
...
+          this.transactionDetails.deliveryType = deliveryType;
+          this.setGooglePaymentRequestConfig(deliveryType);
+
+          return this.currentProductService.getProduct().pipe(
+            take(1),
+            switchMap((product: Product | null) => {
+              const count = this.itemCounterService.getCounter();
+              this.transactionDetails.product = product as Product;
+              this.transactionDetails.quantity = count;
+              this.transactionDetails.context = OpfQuickBuyLocation.PRODUCT;
```
Comment (FollowTheFlo) [inline integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts]: i suppose we can remove this line as it is now also on line 256.

File: integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    return this.currentProductService.getProduct().pipe(
+      take(1),
+      switchMap((product: Product | null) =>
+        this.intendedPickupLocationFacade
+          .getIntendedLocation(product?.code as string)
+          .pipe(
+            map((intendedLocation) => {
+              return intendedLocation?.pickupOption === 'pickup'
```
Comment (FollowTheFlo) [inline integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts]: might need to put 'pickup' in an enum. That will be catch by sonar.

File: integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts [hunk 3]
```diff
@@ -98,8 +101,42 @@ export class OpfGooglePayService {
     },
   };
 
+  private googlePaymentRequest = this.initialGooglePaymentRequest;
+
   protected transactionDetails = this.initialTransactionDetails;
 
+  updateGooglePaymentClient(): void {
```
Comment (FollowTheFlo) [inline integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts]: should we set it as 'protected' or 'private' scope as only used in this class?

File: integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts [hunk 4]
```diff
@@ -98,8 +101,42 @@ export class OpfGooglePayService {
...
+    this.googlePaymentClient = new google.payments.api.PaymentsClient(
+      this.googlePaymentClientOptions
+    );
+  }
+
+  setGooglePaymentRequestConfig(deliveryType: OpfQuickBuyDeliveryType) {
```
Comment (FollowTheFlo) [inline integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts]: same as above

File: integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts (hunk 2 above)
Comment (Matejk00) [reply integration-libs/opf/base/core/services/opf-pickup-in-store-handler.service.ts]: I guess `OpfQuickBuyDeliveryType.PICKUP.toLowerCase()` should do the trick. Pickup option enum is not exposed from pickup in store lib.

**Review by FollowTheFlo — APPROVED** [review]
good to me.

**Pull request merged**